src/Accounts/Authentication/Config/Helper/AppliesToHelper.cs

```diff
@@ -0,0 +1,99 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.PowerShell.Common.Config;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.Commands.Common.Authentication.Config
+{
+    /// <summary>
+    /// Helper class to deal with AppliesTo (how large is the scope that the config affects Azure PowerShell).
+    /// </summary>
+    public static class AppliesToHelper
+    {
+        internal static readonly Regex ModulePattern = new Regex(@"^az\.[a-z]+$", RegexOptions.IgnoreCase);
+        internal static readonly Regex CmdletPattern = new Regex(@"^[a-z]+-[a-z]+$", RegexOptions.IgnoreCase);
```

Comment:
Cmdlet may contains digital number. such as `Get-AzDataFactoryV2`

Reply:
Added numbers to the regex and moved the logic of handling regex into a new `PSNamingUtilities` class.